### PR TITLE
try to reduce beachball on firefox

### PR DESCRIPTION
**Pull request by sandlerr**
cc @zendesk/samson randomly paused the debugger in here a couple of times, but maybe all the message handlers need to be throttled.

### CODE REVIEW

**Comment by grosser** [comment]
:+1: ❤️

**Comment by sandlerr** [comment]
@grosser tried to fix a flaky test, does this look reasonable?

File: test/models/terminal_executor_test.rb [hunk 1]
```diff
@@ -65,13 +65,13 @@
 
     it "terminates the program on getpgid failures so we fail fast" do
       Process.expects(:getpgid).raises(Errno::ESRCH)
-      subject.execute!('echo "hi"').must_equal(nil) # status? of a killed process is 9
+      subject.execute!('sleep 1').must_equal(nil) # status? of a killed process is 9
     end
 
-    it "terminates the program on getpgid + kill failures when the program was already dead" do
+    it "handles the case when getpgid + kill both fail" do
```
Comment (grosser) [inline test/models/terminal_executor_test.rb]: I like the original better `handles` is always weird :D 
`it "kills the program when xyz"`

**Comment by grosser** [comment]
👍 prefer `sleep 0.1` ... we have enough slow tests already :D

File: test/models/terminal_executor_test.rb (hunk 1 above)
Comment (sandlerr) [inline test/models/terminal_executor_test.rb]: I used `handles` to mean `doesn't crash`

Comment (grosser) [inline test/models/terminal_executor_test.rb]: 👍

**Comment by sandlerr** [comment]
ok, I'm done now even though the tests flaked again it looks like it was mysql's fault. We should probably not talk to the db so much in tests. Can I get one more +1?

**Comment by grosser** [comment]
👍

**Pull request merged**